### DISCUSSION TITLE
Is there a documentation somewhere available?

### DISCUSSION THREAD

**Discussion by llacroix** [hannobraun/fornjot#2040] [General]
I checked on crates.io and there is no link to a hosted documentation for the project. That would be very useful to search for things that can be done with the library.

**Comment by hannobraun** [comment]
Hey @llacroix! Which crate where you looking at, specifically? All the crates *should* have documentation linked on `crates.io`, but I've noticed that `fj` doesn't, due to a `docs.rs` build failure: https://docs.rs/crate/fj/0.47.0/builds

If you know of any other crates that don't work, please let me know. I'll go over all of them and look into that build failure once I get a chance.

**Reply by llacroix** [reply]
I think the doc is missing for some but I think I found what I wanted. I wanted to know if there was a way to find to select a face from a solid object.

**Reply by hannobraun** [reply]
I've now looked into the build failures. Looks like the problem was caused by a bug in the nightly version of Rust that was used for the builds. I've opened issues to request rebuilds of the affected crates: https://github.com/rust-lang/docs.rs/issues/2251, https://github.com/rust-lang/docs.rs/issues/2252, https://github.com/rust-lang/docs.rs/issues/2253

**Reply by hannobraun** [reply]
> I wanted to know if there was a way to find to select a face from a solid object.

I just remembered, there have been recent changes here since the last release. You can access the faces of a solid via [`Solid::shells`](https://github.com/hannobraun/fornjot/blob/70fcd955773b73c5bba4c159469ec4ccee754ca8/crates/fj-core/src/objects/kinds/solid.rs#L27-L30) and [`Shell::faces`](https://github.com/hannobraun/fornjot/blob/70fcd955773b73c5bba4c159469ec4ccee754ca8/crates/fj-core/src/objects/kinds/shell.rs#L20-L23) (I think that part hasn't changed since the last release). `Shell::faces` returns a [`Handles<Face>`, which provides various methods to select a specific face](https://github.com/hannobraun/fornjot/blob/main/crates/fj-core/src/objects/handles.rs#L64-L126).

**Reply by hannobraun** [reply]
The missing documentation has been rebuilt now. As far as I know, a documentation link is now available on `crates.io` for all active crates.
